TideApp/TideApp/Scenes/TideTimesView/TideTimesView.swift

```diff
@@ -9,42 +9,33 @@ import SwiftUI
 import Combine
 import CoreLocation
 
-struct TideTimesView: View {
+struct TideTimesView<ViewModel: TideTimesViewModelType>: View {
```

Comment:
Curiosity, is there a reason to have this instead of just make the observedobject the correct viewmodeltype?

Reply:
When I tried `@ObservedObject private var viewModel: TideTimesViewModelType`, the compiler complained with "Protocol 'TideTimesViewModelType' can only be used as a generic constraint because it has Self or associated type requirements"

Reply:
Ahhh ok, definiteley something I really wasn't aware of :D :D